apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -93,26 +93,26 @@ def open(
 
     if not soma_type:
         raise DoesNotExistError(f"{uri!r} does not exist")
+    
+    soma_type = soma_type.lower()
```

Comment:
@nguyenv why?

The "real names" in metadata are camelcased and I fear the lowercasing, which leads to two different case-sensitive spellings now, will be error-prone later on.

Why not simply keep the agreed-on, standard, single camel-cased spellings and use them consistently and error-free everywhere?

Reply:
https://github.com/single-cell-data/TileDB-SOMA/blob/viviannguyen/ndarray-write-path/apis/python/tests/test_factory.py#L41-L78

I modified this because the pre-existing unit tests actually do explicitly check that the type is case insensitive (the d in `NDArray` may be lower or upper case).

Reply:
:( @nguyenv I really don't like mixing cases :( -- would it be too much to ask for us to simply take this opportunity to keystroke out correct casings for once and for all?

Reply:
@nguyenv I really don't like this at all but it appears not to be new to this PR. Thanks.